src/Components/GameScreenComponents/WinningLogic.ts

```diff
@@ -17,180 +17,109 @@ export enum gameResult{
-            {
-                if(gameBoard[trueColumn+i][playerCoordinates.Row] !== playerNumber)
-                {
-                    leftMatches = false;
-                }
-            }
-        }
-
-        if(gameBoard[trueColumn][playerCoordinates.Row] !== playerNumber){
-            leftMatches = false;
+            leftMatches++;
         }
     }
 
-    return (rightMatches || leftMatches);
+    return (leftMatches + rightMatches >= 4);
 }
 
 function checkVerticalWin(playerNumber:number, playerCoordinates:{Column:number, Row:number}, gameBoard:number[][]){
 
-    if(playerNumber === 0)
-    {
-        return false;
-    }
-
-    let topMatches = true;
-    let bottomMatches = true;
+    let topMatches = 0;
+    let bottomMatches = 0;
 
-    /*Check all tokens at the bottom match the token placed...*/
+    //Go 4 tokens to the bottom...
     for(let i = 0; i < 4; i++)
     {
-        let trueRow = playerCoordinates.Row+i
+        let trueRow = playerCoordinates.Row + i;
 
-        if(trueRow > numOfRows-1)
+        if(trueRow > numOfRows - 1)
         {
             break;
         }
 
-        if(trueRow === numOfRows-1)
+        if(gameBoard[playerCoordinates.Column][trueRow] === playerNumber)
         {
-            for(let i = 0; i < 4; i++)
-            {
-                if(gameBoard[playerCoordinates.Column][trueRow-i] !== playerNumber)
-                {
-                    bottomMatches = false;
-                }
-            }
-        }
-
-        if(gameBoard[playerCoordinates.Column][trueRow] !== playerNumber){
-            bottomMatches = false;
+            bottomMatches++;
         }
     }
 
-    /*Check all token at the top match the token placed...*/
-    for(let i = 0; i < 4; i++)
+    //Go 4 tokens to the top...but skip current token position as its already been accounted for
+    for(let i = 1; i < 4; i++)
     {
-        let trueRow = playerCoordinates.Row-i
+        let trueRow = playerCoordinates.Row - i;
 
         if(trueRow < 0)
         {
             break;
         }
 
-        if(trueRow === 0)
+        if(gameBoard[playerCoordinates.Column][trueRow] === playerNumber)
         {
-            for(let i = 0; i < 4; i++)
-            {
-                if(gameBoard[playerCoordinates.Column][trueRow+i] !== playerNumber)
-                {
-                    topMatches = false;
-                }
-            }
-        }
-
-        if(gameBoard[playerCoordinates.Column][trueRow] !== playerNumber){
-            topMatches = false;
+            topMatches++;
         }
     }
 
-    return(topMatches || bottomMatches);
+    return(topMatches + bottomMatches >= 4);
+
 }
 
 function checkDiagonalWin(playerNumber:number, playerCoordinates:{Column:number, Row:number}, gameBoard:number[][]){
 
-    if(playerNumber === 0)
-    {
-        return false;
-    }
-
     /*could be diagonal rightBottom, rightTop, leftBottom, leftTop*/
     let rightBottom = true;
     let rightTop = true;
```

Comment:
Just to check, have you tried testing whether a diagonal line with the winning fourth token placed in the middle is currently detected as a win? Even if it is detected correctly, I think using a similar "right matches" and "left matches" approach would simplify the code here